internal/services/oracle/oracle_autonomous_database_cross_region_disaster_recovery_resource.go

```diff
@@ -0,0 +1,469 @@
+					DataBaseType:                   autonomousdatabases.DataBaseTypeCrossRegionDisasterRecovery,
+					DataStorageSizeInTbs:           pointer.To(model.DataStorageSizeInTbs),
+					DbWorkload:                     pointer.To(autonomousdatabases.WorkloadType(model.DbWorkload)),
+					DbVersion:                      pointer.To(model.DbVersion),
+					DisplayName:                    pointer.To(model.DisplayName),
+					IsAutoScalingEnabled:           pointer.To(model.AutoScalingEnabled),
+					IsAutoScalingForStorageEnabled: pointer.To(model.AutoScalingForStorageEnabled),
+					IsMtlsConnectionRequired:       pointer.To(model.MtlsConnectionRequired),
+					LicenseModel:                   pointer.To(autonomousdatabases.LicenseModel(model.LicenseModel)),
+					NcharacterSet:                  pointer.To(model.NationalCharacterSet),
+					SubnetId:                       pointer.To(model.SubnetId),
+					VnetId:                         pointer.To(model.VnetId),
+				},
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AutonomousDatabaseCrossRegionDisasterRecoveryResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.AutonomousDatabases
+			id, err := autonomousdatabases.ParseAutonomousDatabaseID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %q: %+v", metadata.ResourceData.Id(), err)
+			}
+
+			var model AutonomousDatabaseCrossRegionDisasterRecoveryResourceModel
+			if err = metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			_, err = client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			update := &autonomousdatabases.AutonomousDatabaseUpdate{
+				Properties: &autonomousdatabases.AutonomousDatabaseUpdateProperties{},
+			}
+			if metadata.ResourceData.HasChange("tags") {
+				update.Tags = pointer.To(model.Tags)
+			}
+			if metadata.ResourceData.HasChange("data_storage_size_in_tbs") {
+				update.Properties.DataStorageSizeInTbs = pointer.To(model.DataStorageSizeInTbs)
+			}
+			if metadata.ResourceData.HasChange("compute_count") {
+				update.Properties.ComputeCount = pointer.To(model.ComputeCount)
+			}
+			if metadata.ResourceData.HasChange("auto_scaling_enabled") {
+				update.Properties.IsAutoScalingEnabled = pointer.To(model.AutoScalingEnabled)
+			}
+			if metadata.ResourceData.HasChange("auto_scaling_for_storage_enabled") {
+				update.Properties.IsAutoScalingForStorageEnabled = pointer.To(model.AutoScalingForStorageEnabled)
+			}
+
+			err = client.UpdateThenPoll(ctx, *id, *update)
+			if err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
```

Comment:
id is a pointer so we should dereference it

```suggestion
				return fmt.Errorf("updating %s: %+v", *id, err)
```